### ISSUE TITLE
[BUG] VM has a very serious cache problem. `2.15.0/2.15.3 BETA`

### ISSUE THREAD

**Issue by F9y4ng** [violentmonkey/violentmonkey#1899]
### Problem: 
When I visit a site that has the following code in its HEAD tag:
```html
<meta http-equiv="pragma" content="no-cache">
<meta http-equiv="Cache-Control" content="no-cache, must-revalidate">
<meta http-equiv="expires" content="0">
```
#### ERROR CASE
there is a problem with the VM's cache and an error occurs with GM_getValue, but the value is successfully written via GM_setValue. 
**e.g.,** the problem occurs when I visit https://www.ithome.com,  and I need to either execute the menu command twice or manually refresh the page after executing the menu command once in order to make it work.

![no-cache-site](https://github.com/violentmonkey/violentmonkey/assets/15855068/da81688d-0f8a-45ce-be26-1e4f990b0800)

#### NORMAL CASE
If the site does not have any of the above code in its HEAD tag, it's normal. 

![cached-site](https://github.com/violentmonkey/violentmonkey/assets/15855068/41be309a-3b00-4e87-b06f-0b93ee0d7045)

**It took me a long time to find out and reproduce the bug.**

#### Test Code:
```js
// ==UserScript==
// @name         Test
// @namespace    Test
// @version      0.1
// @description  Test cache
// @author       You
// @match        *://*/*
// @grant        GM_setValue
// @grant        GM_getValue
// @grant        GM_registerMenuCommand
// @run-at       document-start
// ==/UserScript==

(function () {
  "use strict";
  const val = GM_getValue("test");
  console.log("test value:", val);
  if (val === "true") {
    GM_registerMenuCommand(`SET FALSE`, () => {
      GM_setValue("test", "false");
      reload();
    });
  } else {
    GM_registerMenuCommand(`SET TRUE`, () => {
      GM_setValue("test", "true");
      reload();
    });
  }

  function reload() {
    location.reload();
  }
})();

```

#### Environment:
- OS: Windows 10
- Browser: Chrome 117.0.5938.88
- Violentmonkey Version: 2.15.3 BETA / 2.15.0

---

**By the way:** 
When editing code locally in VM, there is a very small chance that the latest modified code will not load, no matter how much you refresh the page, and the browser must be restarted to recover it. I've noticed this problem since version 2.14.0, but haven't been able to find out a way to reproduce it 100% or what is causing the problem.

I think this is another local caching issue in the VM.

**Comment by tophf** [comment]
The script caching problem was fixed in #1793.

**Comment by F9y4ng** [comment]
> The script caching problem was fixed in #1793.

Thanks.
